.github/workflows/pr-opened-command.yml

```diff
@@ -0,0 +1,116 @@
+
+            // Greet contributor
+            console.log('Executing Greet contributor script');
+
+            let firstTimeContributor = true;
+            let message = '';
+
+            const opts = github.issues.listForRepo.endpoint.merge({
+              owner: owner,
+              repo: repo,
+              creator: creator,
+              state: 'all'
+            });
+
+            const issues = await github.paginate(opts);
+
+            for (const issue of issues) {
+              if (issue.number === issueNumber) {
+                continue;
+              }
+
+              if (issue.pull_request) {
+                firstTimeContributor = false;
+                break;
+              }
+            }
+
+            if (firstTimeContributor) {
+              console.log('PR creator is a first time contributor');
+
+              message = `
+              Welcome @${creator}
+
+              Thank you for submitting this PR! If you haven't already [joined our slack community](https://slack.sweetops.com), then we invite you to do so.
+
+              We receive an overwhelming number of contributions. By joining our slack, we'll be able to review your PR faster.
+
+              [![Slack Community](https://slack.cloudposse.com/badge.svg)](https://slack.cloudposse.com)
+              `;
+            }
+            else {
+              console.log('PR creator is already a contributor');
+
+              message = `
+              @${creator} thanks again for your contribution!
+
+              We will review this PR shortly.
+              `;
+            }
+
+            await github.issues.createComment({
+              issue_number: issueNumber,
+              owner: owner,
+              repo: repo,
+              body: message
+            });
+
+            // Assign reviewers
+            console.log('Executing Assign reviewers script');
+
+            const all_reviewers = ['osterman', 'aknysh', 'goruha'];
+
+            // Exclude the PR author from the reviewer list
+            const reviewers = all_reviewers.filter((val) => val !== creator);
+
+            await github.pulls.createReviewRequest({
+              pull_number: issueNumber,
+              owner: owner,
+              repo: repo,
+              reviewers: reviewers
+            });
+
+            await github.issues.addAssignees({
+              issue_number: issueNumber,
+              owner: owner,
+              repo: repo,
+              assignees: [creator]
+            });
+
+            // Assign labels
+            console.log('Executing Assign labels script');
+
+            const changedFilesResponse = await github.pulls.listFiles({
+              pull_number: issueNumber,
+              owner: owner,
+              repo: repo
+            });
+
+            // Max length of labels is 50 chars
+            const changedFiles = changedFilesResponse.data.map(f => f.filename.substring(f.filename.length - 50));
```

Comment:
What's this about? Looks like it adds a label for every file that was changed? This is not the behavior from before. =)

Reply:
Basically, in a monorepo, we want to show which services were modified, not the files.

Reply:
yes
But otherwise, we have to create a mapping b/w labels and file names blobs, which would be different for many repos. Not sure how to handle that from one code base

Reply:
Why is this logic in multiple places? https://github.com/cloudposse/actions/pull/17/files#diff-1770a0951dac10c4b742b621dddbd684R30

Reply:
it works ok
we can limit the label count to let's say max 50

Reply:
the objective is not to show how to add labels or add labels of files. The objective is to add labels when part of the codebase changes. 

> But otherwise, we have to create a mapping b/w labels and file names blobs, which would be different for many repos. Not sure how to handle that from one code base

That's how the other action worked. There was a configuration file. That configuration file could be different per repo. We need to preserve that functionality. The auto-labeling PR is one my favorite that we use and this change would cripple it to something that just adds repo label spam.

Reply:
> Why is this logic in multiple places? https://github.com/cloudposse/actions/pull/17/files#diff-1770a0951dac10c4b742b621dddbd684R30

those are diff workflows
the first one `greet-contributor` is triggered from a comment `\greet-contributor` manually - and we don't really need it if we will have the other one, `pr-opened-command`, which is triggered automatically on PR open and dispatches to `pr-opened-command` workflow which does three things:

* greets the contributor
* assigns labels
* assigns revivers

The same three things could be done by the manual slash command dispatches IF we want to be able to do it

Reply:
> the objective is not to show how to add labels or add labels of files. The objective is to add labels when part of the codebase changes.
> 
> > But otherwise, we have to create a mapping b/w labels and file names blobs, which would be different for many repos. Not sure how to handle that from one code base
> 
> That's how the other action worked. There was a configuration file. That configuration file could be different per repo. We need to preserve that functionality. The auto-labeling PR is one my favorite that we use and this change would cripple it to something that just adds repo label spam.

How do we add the configuration file to each repo?
especially if the config is diff for diff repos?

Reply:
if the config (mapping) is the same, we can implement it in our code w/o copying the 3-rd party code, it's just a few lines of code

Reply:
> How do we add the configuration file to each repo?
especially if the config is diff for diff repos?

That's the responsibility of the repo. It's not the responsibility of the action. 

Here's an example: https://github.com/cloudposse/packages/blob/master/Makefile#L40-L44

Here's another: https://github.com/cloudposse/build-harness/blob/master/Makefile#L18-L23

Note, `.github/auto-label.yml` is *not* a workflow. It's a configuration file. That's why every repo can manage it's own config.

Reply:
and as I mentioned, all 3-rd party actions will not work on dispatch commands since they look for content (repo/owner/PR number) from the event which dispatch commands don't provide.
All of those actions only work on the repo where they are triggered (e.g. by PR open), not dispatched

Reply:
because of that, they could not be in one central repository, but will have to be added to every other repo

Reply:
> > How do we add the configuration file to each repo?
> > especially if the config is diff for diff repos?
> 
> That's the responsibility of the repo. It's not the responsibility of the action.
> 
> Here's an example: https://github.com/cloudposse/packages/blob/master/Makefile#L40-L44
> 
> Here's another: https://github.com/cloudposse/build-harness/blob/master/Makefile#L18-L23
> 
> Note, `.github/auto-label.yml` is _not_ a workflow. It's a configuration file. That's why every repo can manage it's own config.

I understand that.
But adding diff configs for 200-300 repos is not a fun task.
Having the code/logic to map it in one place would be much better

Reply:
we can remove the assign-label logic from the code and just use the other actions for now.
We need rebuild README and `terraform fmt` in almost every PR

Reply:
Right, I understand why you rewrote it. That makes sense. My point is the business logic is not the same and we need to preserve the functionality, meaning the ability to specify labels in a config file. The config file behavior not implemented by GitHub (I think), it's implemented by the author of the action.

Reply:
> we can remove the assign-label logic from the code and just use the other actions for now.

Okay, let's do that.

Reply:
we can copy that logic into our code, it's 10 lines of code.
And we then can checkout the branch to read the config file and then apply labels